pkg/core/manager/workflow.go

```diff
@@ -72,6 +72,31 @@ func (mgr *workflowManager) AssignToCodeset(ctx context.Context, name, codesetPr
 }
 
 func (mgr *workflowManager) UnassignFromCodeset(ctx context.Context, name, codesetProject, codesetName string) (err error) {
+	codeset, err := mgr.codesetStore.Find(ctx, codesetProject, codesetName)
+	if err != nil {
+		return err
+	}
+
+	assignment, err := mgr.workflowStore.GetAssignedCodeset(ctx, name, codeset)
+	if err != nil {
+		return err
+	}
+
+	if assignment.WebhookID != nil {
+		err = mgr.codesetStore.DeleteWebhook(ctx, codeset, assignment.WebhookID)
+		if err != nil {
+			return err
```

Comment:
Same thing here, only in reverse: shouldn't you collect the error and continue to delete the listener instead of failing fast ?

Reply:
If we do that we leave the webhook on the codeset. Failing here allows a retry for deleting as the assignment still exists.

Reply:
Ok, that's one way of looking at it.